base/exports.jl

```diff
@@ -1063,3 +1063,67 @@ export
     @view,
     @views,
     @static
+
+# TODO: use normal syntax once JuliaSyntax.jl becomes available at this point in bootstrapping
+eval(Expr(:public,
+    Symbol("@assume_effects"),
+    Symbol("@constprop"),
+    Symbol("@locals"),
+    Symbol("@propagate_inbounds"),
+    :AbstractLock,
+    :AsyncCondition,
+    :CFunction,
```

Comment:
There shouldn't be much need to access this directly, but I suppose `@cfunction` does so indirectly into user code?

Reply:
```suggestion
```